graphql-server/src/middlewares/session.ts

```diff
@@ -0,0 +1,65 @@
+import { Express } from "express";
+import { Strategy } from "openid-client";
+const passport = require("passport");
+const session = require("express-session");
+import { buildProps } from "../utils/buildProps";
+import { getKeycloakClient } from "../utils/session";
+
+const props = buildProps();
+
+/**
+ * This middleware sets up the following session-related mechanisms:
+ * - Express-session for storing user info in session store and managing the session cookie
+ * - Passport for authenticating users using OpenID Connect (OIDC) protocol with Keycloak as the OIDC provider
+ * - Storing timestamp of user's last activity for the auto-timeout functionality
+ */
+module.exports = async function (app: Express) {
```

Comment:
best to give a function like this a name that describes what it does (e..g createSession)